src/odemis/acq/align/test/align_test.py

```diff
@@ -0,0 +1,203 @@
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+from concurrent import futures
+from concurrent.futures._base import CancelledError
+import logging
+from odemis import model
+
+from numpy.core.multiarray import ndarray
+
+from odemis.acq.align import turnLightAndCheck
+from odemis.driver import simulated
+from odemis.dataio import tiff, hdf5
+from odemis.util import mock, img
+from odemis.util import TimeoutError
+import os
+import time
+import unittest
+import numpy
+import cv2
+
+
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)-15s: %(message)s")
+logging.getLogger().setLevel(logging.DEBUG)
+
+
+TEST_IMAGE_PATH = os.path.dirname(__file__)
+
+class TestTurnLightAndCheck(unittest.TestCase):
+    """
+    Test turnLightAndCheck functions
+    """
+
+    def setUp(self):
+        self.light = simulated.Light("Calibration Light", "brightlight")
+
+        self.data = tiff.read_data(os.path.join(TEST_IMAGE_PATH, "brightlight-off-slit-spccd-simple.ome.tiff"))
+        self.img_light_off = img.ensure2DImage(self.data[0])
+
+    def test_slit_off_on(self):
+        """
+        Test from dark slit image to bright slit image (gray)
+        """
+        bl = self.light
+        ccd = mock.FakeCCD(0, self.img_light_off)
+        f = turnLightAndCheck(bl, ccd)
+        time.sleep(1)
+        self.assertFalse(f.done())
+
+        # simulate light turning on
+        self.data = tiff.read_data(os.path.join(TEST_IMAGE_PATH, "brightlight-on-slit-spccd-simple.ome.tiff"))
+        img_light_on = img.ensure2DImage(self.data[0])
+
+        ccd.fake_img = img_light_on
+        time.sleep(1)
+        self.assertTrue(f.done())
+        f.result()
+
+ #    @skip("skip")
+    def test_slit_timeout(self):
+        """
+        Test Timeout
+        """
+        bl = self.light
+        ccd = mock.FakeCCD(0, self.img_light_off)
+        f = turnLightAndCheck(bl, ccd)
+        # no change in the image
+        time.sleep(1)
+        self.assertFalse(f.done())
+        # with self.assertRaises(TimeoutError):
+        #     f.result(timeout=5)
+        with self.assertRaises(futures.TimeoutError):
+            f.result(timeout=5)
+        f.cancel()
+        logging.warning("Light doesn't appear to turn on after 60s")
+
+
+    def test_slit_cancel(self):
+        """
+        Test cancelling does cancel (relatively quickly)
+        """
+        bl = self.light
+        ccd = mock.FakeCCD(0, self.img_light_off)
+        f = turnLightAndCheck(bl, ccd)
+        time.sleep(1)
+        f.cancel()
+        self.assertTrue(f.cancelled())
+        with self.assertRaises(CancelledError):
+            f.result(5)
+
+    def test_1pixel_image(self):
```

Comment:
Essentially, it's like `test_slit_off_on()`, but with a different image. Please name it something like `test_1pix_off_on()`.